catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -109,6 +118,45 @@ public ContentDirectoryMonitor(CamelContext camelContext, int maxRetries,
     }
 
     /**
+     * Set the thread pool size with the given argument.  If the given argument is less than 1,
+     * numThreads is set to 1.  If the given argument is greater than the MAX_THREAD_SIZE, numThreads
+     * is set to MAX_THREAD_SIZE.
+     *
+     * @param numThreads - the specified size to make the thread pool
+     */
+    public void setNumThreads(Integer numThreads) {
+        if (numThreads < 1) {
+            this.numThreads = 1;
+        } else if (numThreads > MAX_THREAD_SIZE) {
+            this.numThreads = MAX_THREAD_SIZE;
+        } else {
+            this.numThreads = numThreads;
+        }
+    }
+
+    public Integer getNumThreads() {
+        return numThreads;
+    }
+
+    /**
+     * Set the read lock interval for the Camel Route with the given argument.  If the readLockIntervalMilliseconds
+     * is less than 100, set it to 100.
+     *
+     * @param readLockIntervalMilliseconds
+     */
+    public void setReadLockIntervalMilliseconds(Integer readLockIntervalMilliseconds) {
+        if (readLockIntervalMilliseconds < MIN_READLOCK_INTERVAL_MILLISECONDS) {
```

Comment:
you could use the same Math.min/Math.max trick:

this.readLockIntervalMilliseconds = Math.max(readLockIntervalMilliseconds, MIN_READLOCK_INTERVAL_MILLSECONDS);